### ISSUE TITLE
RC 0.5.0.0: "Like"-Action triggers subscription of the post

### ISSUE THREAD

**Issue by Faldrian** [diaspora/diaspora#5881]
When testing the RC 0.5.0.0 several people noticed: You subscribe to the post when you like the post.
This is different then before - previous to RC 0.5.0.0 you would only subscripte to a post when you comment on the post.

I don't know if this is intentional, but it is an important change in user interaction design and if intentional it should be discussed (maybe it has been, then sorry for not noticing).
I often "like" posts just flying by and don't want to have my notifications spammed by default. I also often think quite a lot before commenting: "Do I want to add my 2 cents and risk to get the whole load of notifications about comments which are uninteresting because I had that discussion several times... or do I just like it and be done with it."

So if this is intentional, the default should be configurable - or it should be changed to "do not subscribe on like" like it was in previous versions.

**Comment by jhass** [comment]
A bit background: The subscription is based on what we call participations. A like always created a participation, so did a comment or a reshare. The new "subscribe to post" feature works by creating a participation but no other interaction with the post. The list of people to notify is determined by who has a participation for that post, previously we checked whether a comment by the recipient existed before creating the notification, but that wouldn't work for the "subscribe to post" feature. So it could be said the new behavior is a compromise to be able to implement the "subscribe to post" feature without a large redesign of the codebase or adding a second huge table to manage them explicitly.

This was changed in #5511

**Comment by Faldrian** [comment]
Thanks a lot for the explaination!
Maybe it will turn out a good feature to have yourself notified about comments on post you did only "like". Maybe not... at least the behaviour is more consistent now from a programmers point of view.

As diaspora is a changing community-project, we might try this and make adjustments later... sorry for the "boa, things changed, I have to make a big deal of it" issue - but if shows how we need to have a good "user transition guide" or something to show people what changed in the new release, so they are not as surprised and will not start thinking of these things as bugs. :)

**Comment by jhass** [comment]
Well, thinking about it, we could easily add a flag in the participations table whether notifications for it should be generated or not, it's just our biggest table, so even that bit for the flag will likely blow it up quite a bit further.

**Comment by Faldrian** [comment]
Hm... I think "make good defaults and not so many settings" may have been one of the design principles we tend to use... otherwise I would say it could be simple to just generate a participation when clicking "like" and having a "follow on like" option (per user) enabled in the settings. That would not have something to do with the participations table but more with the user table which has a lot less rows.

**Comment by jhass** [comment]
No, the main purpose of the participations table is to track which users (and thus pods) should receive interactions with a post, the notifications are only the second step after that.

**Comment by Faldrian** [comment]
I understand that. If a user is not added to the participations table (for that post) the user will not receive interactions. If I like a post but will not be added to the table, the behaviour will be as it was in the old version. Whether someone is added to the table or not could be controlled by a user-specific setting that is evaluated when the user clicks the like-button.

**Comment by Flaburgan** [comment]
Note that now you can also unsubscribe to notification, so no need to think a lot about "do I have to comment and be spammed", you can comment / like and immediately stop receiving notification for the post ;)

**Comment by Faldrian** [comment]
@Flaburgan I know that. I just don't want to hit "unsubscribe" 100x when I like 100 posts where I don't want any notifications and "just" liked them.

**Comment by rebeka-catalina** [comment]
@Flaburgan 
Yes, that's really a good feature, absolutely agree. I also see now a bit the background of the table and coding "problems".
But from in practical view on "the other side" it's not done with disabling the notifications - that's also only the second step. If I like a post by clicking, then in background the subscription already happens, but the button appears as disabled. So we have to refresh the page and scroll back, or to open the post itself, to make the notifications-button usable to disable them.

**Comment by Flaburgan** [comment]
@rebeka-catalina this corresponds to the issue I open: https://github.com/diaspora/diaspora/issues/5811

Btw, maybe we can add a small fix to this problem by checking when someone likes a post if there was no participation before. In this case, we call the unsubscribe action right after the like.

**Comment by rebeka-catalina** [comment]
Aah ok - thank you. Didn't see that :)

**Comment by Faldrian** [comment]
Hm... as far as I can see this behaviour proves to be quite acceptable in daily use. I'd like to withdraw my bugreport. ;)

**Comment by jhass** [comment]
Reopen because of #6018

**Comment by elektrollart** [comment]
A other problem with this function is, the like button have the same function as the subscription, the only difference is, people see the feedback. As a user, i like less than before, cause I'm not in mood to unsubscribe the posts i liked. Now I like only the posts, I want to comment. I talked with other users about this and some people see it the same way like me, other dislike this function, cause they don't want to unsubscribe all posts they like. A other idea is, to throw away the like button, cause it works like a subscription.

**Comment by rebeka-catalina** [comment]
Perhaps it's a very bad idea in technical, I'm not sure so I simply ask : 
Could it be a possibility to create a per-user option like the nsfw-option ?

I also really don't like it, how it is actually, because it's like elektrollart already described : 
4 clicks are needed just to leave a like. There are some people, who say they like it, but also quite much people, who reduce their activities, because of the behavior of the like-action.

**Comment by Globulle** [comment]
I agree with @elektrollart and @rebeka-catalina . The "like" action should remain a really easy one, without other consequence than notifying the author and incrementing the "like" count. You bring kind of support but you don't want to get involved in the discussion, otherwise you generally comment. I think there is a kind of hierarchy among the actions we have to preserve for the whole participation system to work.

**Comment by Flaburgan** [comment]
I think we need to at least:
- Move the notification icon next to the Like, Comment and Reshare icons in the SPV, and to the bottom of the post next to the links to link, comment and reshare in the stream
- Make its state change immediately when we like / comment the post, to avoid a page refresh. See https://github.com/diaspora/diaspora/issues/5811

This should already really improve the situation. If it's not enough, we could click on the unsubscribe icon with JS when the like icon is pressed, to immediately unsubscribe the user.

**Comment by 4ndyD** [comment]
Add an option in notification settings to toggle whether to automatically subscribe to all posts that you like/comment/reshare or not.

I do want those automatic notifications right now, but might not in the future, and some other people don't want them, so there should be an option to select whichever default value suits you. Title should be changed accordingly to suit more users.

**Comment by SpcCw** [comment]
Having this configurable would be really good. This is annoying for many users.

Clicking the unsubscribe icon right after liking is not always convenient - if the post is rather long and/or monitor resolution is not large then the bell icon gets off screen and user has to scroll back up, click the bell and then resume by scrolling back down. Very clunky experience. And it is impossible to do from touch-optimized (mobile) view.
